content/articles/django-custom-template-tags-and-filters/index.md

```diff
@@ -0,0 +1,258 @@
+### Introduction
+In this tutorial, wea are going to look at how to create custom template tags and filters in Django.
+
+Django template language comes with built-in template tags such as `{% if %}` and filters such as `safe`. You may however find that you require a functionality not included in Django by default. In this case, you create them using Python and avail them using the `load ` tag.
+
+Let’s get started!
+
+###  Prerequisites
+To follow along it's important that you have:
+1. The ablility to create a simple Django application
+2. Python and Django installed.
+3. A suitable IDE such as Pycharm, VS Code etc.
+
+### Takeaways
+By the end of this tutorial, you will be able to:
+1. Understand the use of custom template tags and filters and how they work.
+2. Create your own customized template tags and filters to suit your needs.
+
+### Getting started.
+Your custom tags and filters must be within a Django app. If you want your tags and filters to be used across your entire project, you may create an app just to accomplish that.
+
+1. Create an app called `Contacts` and add it in your `INSTALLED_APPS`. Inside your app , create a directory called `templatetags` in the same level as the views, models etc.
+2. Make the `templatetags` directory a Python package by creating an `__init__.py` file. Create another file `custom_tags.py` to hold your custom tags and filters.
+
+Your app structure will now look something similar to this:
+
+```python
+.Contact
+├── contact
+│   ├── admin.py
+│   ├── apps.py
+│   
+│   ├── migrations
+│   │   └──  __init__.py
+│   │  
+│   │       
+│   ├── models.py
+│   
+│   ├── templatetags
+│   │  └──  custom_tags.py
+│   │
+│   ├── tests.py
+│   ├── url.py
+│   └── views.py
+├── Contact
+│   ├── asgi.py
+│   ├── settings.py
+│   ├── urls.py
+│   └── wsgi.py
+├── db.sqlite3
+└──  manage.py
+```
+
+You can create as many modules as you can inside the `templatetags` folder. You should, however, note that using the `{% load %}` tag will only work for a particular module name, not the app name.
+
+```python
+from Django import template
+register = template.Library()
+```
+
+To load the tags and filters, we will do the following in our template:
+
+```jinja
+{% load custom_tags %}
+```
+
+### Writing Custom Filters
```

Comment:
Capitalize the first word in the heading, not every word in the heading, unless it is a noun. Please do this to all your headings.

Reply:
Done